lib/packages.js

```diff
@@ -467,3 +469,34 @@ function _isDir(filename) {
   } catch(e) {}
   return stat ? stat.isDirectory() : false;
 }
+
+function _detectCycle(cache) {
+  const initialPkgName = Object.keys(cache)[0];
+  const visited = new Set();
+  const recStack = {};
```

Comment:
Could make all of these the default values for the parameters to the function and then just have one function that calls itself rather than two (`_detectCycle` and `_detectCycleUtil`).